src/main/java/com/akathist/maven/plugins/launch4j/Launch4jMojo.java

```diff
@@ -288,52 +296,61 @@ public void execute() throws MojoExecutionException {
-        c.setLibs(relativizeAndCopy(workdir, libs));
-        c.setVariables(vars);
-
-        if (classPath != null) {
-            c.setClassPath(classPath.toL4j(dependencies));
-        }
-        if (jre != null) {
-            c.setJre(jre.toL4j());
-        }
-        if (singleInstance != null) {
-            c.setSingleInstance(singleInstance.toL4j());
-        }
-        if (splash != null) {
-            c.setSplash(splash.toL4j());
-        }
-        if (versionInfo != null) {
-            c.setVersionInfo(versionInfo.toL4j());
-        }
-        if (messages != null) {
-            c.setMessages(messages.toL4j());
+        final File workDir = setupBuildEnvironment();
+        if (infile != null && infile.exists()) {
+            try {
+                if (getLog().isDebugEnabled()) {
+                    getLog().debug("Trying to load Launch4j native configuration using file=" + infile.getAbsolutePath());
+                }
+                ConfigPersister.getInstance().load(infile);
+            } catch (ConfigPersisterException e) {
+                getLog().error(e);
+                throw new MojoExecutionException("Could not load Launch4j native configuration file", e);
+            }
+        } else {
+            final Config c = new Config();
+
+            c.setHeaderType(headerType);
+            c.setOutfile(outfile);
+            c.setJar(getJar());
+            c.setDontWrapJar(dontWrapJar);
+            c.setErrTitle(errTitle);
+            c.setDownloadUrl(downloadUrl);
+            c.setSupportUrl(supportUrl);
+            c.setCmdLine(cmdLine);
+            c.setChdir(chdir);
+            c.setPriority(priority);
+            c.setStayAlive(stayAlive);
+            c.setRestartOnCrash(restartOnCrash);
+            c.setManifest(manifest);
+            c.setIcon(icon);
+            c.setHeaderObjects(relativizeAndCopy(workDir, objs));
+            c.setLibs(relativizeAndCopy(workDir, libs));
+            c.setVariables(vars);
+
+            if (classPath != null) {
+                c.setClassPath(classPath.toL4j(dependencies));
+            }
+            if (jre != null) {
+                c.setJre(jre.toL4j());
+            }
+            if (singleInstance != null) {
+                c.setSingleInstance(singleInstance.toL4j());
+            }
+            if (splash != null) {
+                c.setSplash(splash.toL4j());
+            }
+            if (versionInfo != null) {
+                c.setVersionInfo(versionInfo.toL4j());
+            }
+            if (messages != null) {
+                c.setMessages(messages.toL4j());
+            }
+            ConfigPersister.getInstance().setAntConfig(c, getBaseDir());
         }
 
-        ConfigPersister.getInstance().setAntConfig(c, getBaseDir());
-        Builder b = new Builder(new MavenLog(getLog()), workdir);
-
         try {
-            b.build();
+            new Builder(new MavenLog(getLog()), workDir).build();
```

Comment:
Separating initialisation and execution is a good idea, now you have three potential places where exception can be thrown and such code is harder to debug, could you revert to previous two step version?